api/v1alpha1/project_config_types.go

```diff
@@ -46,6 +86,29 @@ type PromotionPolicy struct {
 	AutoPromotionEnabled bool `json:"autoPromotionEnabled,omitempty" protobuf:"varint,2,opt,name=autoPromotionEnabled"`
 }
 
+// WebhookReceiverConfig describes the configuration for webhook receivers.
+type WebhookReceiverConfig struct {
+	// GitHub contains the configuration for a webhook receiver that is compatible with
+	// GitHub payloads.
+	GitHub *GitHubWebhookReceiver `json:"github,omitempty" protobuf:"bytes,1,opt,name=github"`
```

Comment:
Can you add a todo here about making this mutually exclusive with the configs for other platforms?

Reply:
I suppose we should also, temporarily, have this be a required field.